src/primitives/createLazyLoadQuery.ts

```diff
@@ -1,49 +1,164 @@
-import { access } from "@solid-primitives/utils";
 import type { MaybeAccessor } from "@solid-primitives/utils";
-import { __internal } from "relay-runtime";
-import type {
-	CacheConfig,
-	FetchPolicy,
-	GraphQLTaggedNode,
-	OperationType,
-	RenderPolicy,
-	VariablesOf,
+import RelayRuntime from "relay-runtime";
+import {
+	type CacheConfig,
+	type FetchQueryFetchPolicy,
+	type GraphQLResponse,
+	type GraphQLTaggedNode,
+	Observable,
+	type OperationDescriptor,
+	type OperationType,
+	type VariablesOf,
 } from "relay-runtime";
-import { createMemo } from "solid-js";
-import type { ResourceReturn } from "solid-js";
+import {
+	batch,
+	createComputed,
+	createResource,
+	createSignal,
+	onCleanup,
+} from "solid-js";
+import { createStore, reconcile } from "solid-js/store";
 
 import { useRelayEnvironment } from "../RelayEnvironment";
-import { createLazyLoadQueryNode } from "./createLazyLoadQueryNode";
+import { type DataProxy, makeDataProxy } from "../utils/dataProxy";
 import { createMemoOperationDescriptor } from "./createMemoOperationDescriptor";
 
+type QueryResult<T> = {
+	data: T | undefined;
+	error: unknown;
+	pending: boolean;
+	inFlight: boolean;
+};
+
 export function createLazyLoadQuery<TQuery extends OperationType>(
 	gqlQuery: MaybeAccessor<GraphQLTaggedNode>,
 	variables: MaybeAccessor<VariablesOf<TQuery>>,
 	options?: {
-		fetchKey?: MaybeAccessor<string | number | undefined>;
-		fetchPolicy?: MaybeAccessor<FetchPolicy | undefined>;
+		fetchPolicy?: MaybeAccessor<FetchQueryFetchPolicy | undefined>;
 		networkCacheConfig?: MaybeAccessor<CacheConfig | undefined>;
-		UNSTABLE_renderPolicy?: MaybeAccessor<RenderPolicy | undefined>;
 	},
-): ResourceReturn<TQuery["response"]> {
+): DataProxy<TQuery["response"]> {
 	const environment = useRelayEnvironment();
 
-	const query = createMemoOperationDescriptor(
+	const operation = createMemoOperationDescriptor(
 		gqlQuery,
 		variables,
-		() => access(options?.networkCacheConfig) ?? { force: true },
+		options?.networkCacheConfig,
+	);
+	const [source, setSource] = createSignal<{
+		observable: Observable<GraphQLResponse>;
+		operation: OperationDescriptor;
+	}>();
+	const [serverData, setServerData] = createSignal<TQuery["response"]>();
+
+	const [resource] = createResource(
+		operation,
+		async (operation) => {
+			const observable = RelayRuntime.__internal.fetchQuery(
+				environment,
+				operation,
+			);
+			setSource({ observable, operation });
+			const stream = new ReadableStream<GraphQLResponse>({
+				start(controller) {
+					observable.subscribe({
+						next(response) {
```

Comment:
My Solid knowledge is spotty here, so I'm not sure I follow what's going on here, but I'll try to add some context about Relay.

One thing that's not intuitive here is that the `response` here is a snapshot which is not by itself safe to show to users (`@required` and friends have not been processed). Additionally it does not subscribe to changes after the response ends.

Small fact: You can also use `observeFragment` for queries/mutations. We think of each operation (query/mutation) as having a fragment component which is the data that it reads itself (excluding any fragment spreads). You can see this in the operation's generated `.graphql.js`.

Reply:
What arguments should I pass to `observeFragment`? I tried `observeFragment(env, OperationDescriptor.fragment, {})` and `invariant` got triggered because of `isFragment()` returning false ;(

Reply:
Nevermind, found [this](https://github.com/facebook/relay/blob/276d2dd749105b9bbd9f14dfe2eef8ba60b48d64/packages/react-relay/relay-hooks/QueryResource.js#L105-L111) and got it working 👍